src/zkevm_specs/evm/execution/sdiv_smod.py

```diff
@@ -0,0 +1,132 @@
+
+    pop1 = instruction.stack_pop()
+    pop2 = instruction.stack_pop()
+    push = instruction.stack_push()
+
+    (
+        quotient,
+        divisor,
+        remainder,
+        dividend,
+    ) = gen_witness(opcode, pop1, pop2, push)
+    check_witness(
+        instruction,
+        quotient,
+        divisor,
+        remainder,
+        dividend,
+    )
+
+    instruction.step_state_transition_in_same_context(
+        opcode,
+        rw_counter=Transition.delta(3),
+        program_counter=Transition.delta(1),
+        stack_pointer=Transition.delta(1),
+    )
+
+
+def check_witness(
+    instruction: Instruction,
+    quotient: RLC,
+    divisor: RLC,
+    remainder: RLC,
+    dividend: RLC,
+):
+    quotient_abs, quotient_is_neg = instruction.abs_word(quotient)
+    divisor_abs, divisor_is_neg = instruction.abs_word(divisor)
+    remainder_abs, remainder_is_neg = instruction.abs_word(remainder)
+    dividend_abs, dividend_is_neg = instruction.abs_word(dividend)
+
+    quotient_is_non_zero = 1 - instruction.word_is_zero(quotient)
+    divisor_is_non_zero = 1 - instruction.word_is_zero(divisor)
+    remainder_is_non_zero = 1 - instruction.word_is_zero(remainder)
+
+    # Constrain the ABS words of quotient, divisor, remainder and dividend.
+
+    # Function `mul_add_words` constrains `|quotient| * |divisor| + |remainder| = |dividend|`.
+    overflow = instruction.mul_add_words(quotient_abs, divisor_abs, remainder_abs, dividend_abs)
+    # Constrain overflow == 0.
+    instruction.constrain_zero(overflow)
+
+    # Constrain abs(remainder) < abs(divisor) when divisor != 0.
+    remainder_abs_lt_divisor_abs, _ = instruction.compare_word(remainder_abs, divisor_abs)
+    instruction.constrain_zero((1 - remainder_abs_lt_divisor_abs) * divisor_is_non_zero)
+
+    # Constrain sign(dividend) == sign(remainder) when quotient, divisor and
+    # remainder are all non-zero.
+    condition = quotient_is_non_zero * divisor_is_non_zero * remainder_is_non_zero
+    instruction.constrain_equal(dividend_is_neg * condition, remainder_is_neg * condition)
+
+    # For a special `SDIV` case, when input `dividend = -(1 << 255)` and `divisor = -1`,
+    # the quotient result should be `1 << 255`. But a `signed` word could only express
+    # `signed` value from `-(1 << 255)` to `(1 << 255) - 1`. So below constraint
+    # `sign(dividend) == sign(divisor) ^ sign(quotient)` cannot be applied for this case.
+    dividend_is_signed_overflow = instruction.word_is_neg(dividend_abs)
+
+    # Constrain sign(dividend) == sign(divisor) ^ sign(quotient) when both
+    # quotient and divisor are non-zero and dividend is not signed overflow.
+    condition = quotient_is_non_zero * divisor_is_non_zero * (1 - dividend_is_signed_overflow)
+    instruction.constrain_equal(
+        (1 - dividend_is_neg) * condition,
+        ((quotient_is_neg * divisor_is_neg) + (1 - quotient_is_neg) * (1 - divisor_is_neg))
+        * condition,
+    )
```

Comment:
simplify the constraint

```suggestion
    instruction.constrain_zero(
        (quotient_is_neg + divisor_is_neg - 2 * quotient_is_neg * divisor_is_neg - dividend_is_neg) * condition,
    )
```